lib/philomena/tags.ex

```diff
@@ -732,4 +732,55 @@ defmodule Philomena.Tags do
   def change_implication(%Implication{} = implication) do
     Implication.changeset(implication, %{})
   end
+
+  @doc """
+  Deletes all tags that meet all of the following conditions:
+  - No category assigned
+  - Zero images count
+  - No description
+  - No short description
+  - No mod notes
+  - No image attached
+  - Not aliased to another tag
+  - Has no aliases pointing to it
+  - Does not imply any other tags
+  - Is not implied by any other tags
+
+  Also removes the deleted tags from the search index.
+
+  ## Examples
+
+      iex> cleanup!()
+      {3, [1, 2, 3]}
+
+  """
+  def cleanup! do
+    tag_ids =
+      from(t in Tag,
+        where: is_nil(t.category),
+        where: t.images_count == 0,
```

Comment:
`images_count` field is unreliable, you should query `select count(*) from image_taggings where tag_id = TAG_ID;`. `images_count` will be set to 0 even if the tag IS actually present on some deleted image, which might leave that deleted image with less than 3 tags. This should never happen.